crates/bin/c8y_configuration_poc/src/c8y_http_proxy/actor.rs

```diff
@@ -0,0 +1,420 @@
+    type MessageBox = C8YHttpProxyMessageBox;
+
+    fn name(&self) -> &str {
+        "C8YHttpProxy"
+    }
+
+    async fn run(self, messages: Self::MessageBox) -> Result<(), ChannelError> {
+        C8YHttpProxyActor::run(self, messages).await
+    }
+}
+
+pub(crate) struct C8YHttpProxyMessageBox {
+    /// Connection to the clients
+    pub(crate) clients: ServiceMessageBox<C8YRestRequest, C8YRestResult>,
+
+    /// Connection to an HTTP actor
+    pub(crate) http: HttpHandle,
+
+    /// Connection to a JWT token retriever
+    pub(crate) jwt: JwtRetriever,
+}
+
+#[derive(Debug)]
+pub struct C8YRestRequestWithClientId(usize, C8YRestRequest);
+
+#[derive(Debug)]
+pub struct C8YRestResponseWithClientId(usize, C8YRestResult);
+
+fan_in_message_type!(C8YHttpProxyInput[C8YRestRequestWithClientId, HttpResult, JwtResult] : Debug);
+fan_in_message_type!(C8YHttpProxyOutput[C8YRestResponseWithClientId, HttpRequest, JwtRequest] : Debug);
+
+// TODO: Can such a MessageBox implementation be derived from a struct of message boxes?
+#[async_trait]
+impl MessageBox for C8YHttpProxyMessageBox {
+    type Input = C8YHttpProxyInput;
+    type Output = C8YHttpProxyOutput;
+
+    async fn recv(&mut self) -> Option<Self::Input> {
+        tokio::select! {
+            Some((id,message)) = self.clients.recv() => {
+                Some(C8YHttpProxyInput::C8YRestRequestWithClientId(C8YRestRequestWithClientId (id, message)))
+            },
+            Some(message) = self.http.recv() => {
+                Some(C8YHttpProxyInput::HttpResult(message))
+            },
+            Some(message) = self.jwt.recv() => {
+                Some(C8YHttpProxyInput::JwtResult(message))
+            },
+            else => None,
+        }
+    }
+
+    async fn send(&mut self, message: Self::Output) -> Result<(), ChannelError> {
+        match message {
+            C8YHttpProxyOutput::C8YRestResponseWithClientId(message) => {
+                self.clients.send((message.0, message.1)).await
+            }
+            C8YHttpProxyOutput::HttpRequest(message) => self.http.send(message).await,
+            C8YHttpProxyOutput::JwtRequest(message) => self.jwt.send(message).await,
+        }
+    }
+
+    fn turn_logging_on(&mut self, _on: bool) {
+        todo!()
+    }
+
+    fn name(&self) -> &str {
+        "C8Y-REST"
+    }
+
+    fn logging_is_on(&self) -> bool {
+        true
+    }
+}
+
+impl C8YHttpProxyActor {
+    pub fn new(config: C8YHttpConfig) -> Self {
+        let unknown_internal_id = "";
+        let end_point = C8yEndPoint::new(&config.c8y_host, &config.device_id, unknown_internal_id);
+        let child_devices = HashMap::default();
+        C8YHttpProxyActor {
+            end_point,
+            child_devices,
+        }
+    }
+
+    pub async fn run(mut self, mut messages: C8YHttpProxyMessageBox) -> Result<(), ChannelError> {
+        let clients = &mut messages.clients;
+        let http = &mut messages.http;
+        let jwt = &mut messages.jwt;
```

Comment:
This somehow feels wrong. The actor ideally should just be sending messages to the mailbox and the mailbox internally should be using the `http` and `jwt` handles to do what it's supposed to handle the messages "sent" by this actor, without having to expose them to the consuming actor.